GitHubBranches.ps1

```diff
@@ -154,6 +154,351 @@ filter Get-GitHubRepositoryBranch
+        GitHub.Event
+        GitHub.Issue
+        GitHub.IssueComment
+        GitHub.Label
+        GitHub.Milestone
+        GitHub.PullRequest
+        GitHub.Project
+        GitHub.ProjectCard
+        GitHub.ProjectColumn
+        GitHub.Release
+        GitHub.Repository
+
+    .OUTPUTS
+        GitHub.Branch
+
+    .EXAMPLE
+        New-GitHubRepositoryBranch -OwnerName microsoft -RepositoryName PowerShellForGitHub -TargetBranchName new-branch
+
+        Creates a new branch in the specified repository from the master branch.
+
+    .EXAMPLE
+        New-GitHubRepositoryBranch -Uri 'https://github.com/microsoft/PowerShellForGitHub' -BranchName develop -TargetBranchName new-branch
+
+        Creates a new branch in the specified repository from the 'develop' origin branch.
+
+    .EXAMPLE
+        $repo = Get-GithubRepository -Uri https://github.com/You/YourRepo
+        $repo | New-GitHubRepositoryBranch -TargetBranchName new-branch
+
+        You can also pipe in a repo that was returned from a previous command.
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParameterSetName = 'Elements',
+        PositionalBinding = $false
+    )]
+    [OutputType({$script:GitHubBranchTypeName})]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSShouldProcess', '',
+        Justification = 'Methods called within here make use of PSShouldProcess, and the switch is
+        passed on to them inherently.')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '',
+        Justification = 'One or more parameters (like NoStatus) are only referenced by helper
+        methods which get access to it from the stack via Get-Variable -Scope 1.')]
+    [Alias('New-GitHubBranch')]
+    param(
+        [Parameter(ParameterSetName = 'Elements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName = 'Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            Position = 1,
+            ParameterSetName = 'Uri')]
+        [Alias('RepositoryUrl')]
+        [string] $Uri,
+
+        [string] $BranchName = 'master',
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipeline,
+            Position = 2)]
+        [string] $TargetBranchName,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $elements = Resolve-RepositoryElements
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+    }
+
+    try
+    {
+        $getGitHubRepositoryBranchParms = @{
+            OwnerName = $OwnerName
+            RepositoryName = $RepositoryName
+            BranchName = $BranchName
+            Whatif = $false
+            Confirm = $false
```

Comment:
Because `Get-GitHubRepositoryBranch` currently has `SupportsShouldProcess` enabled (which it shouldn't have), so if a user passes an explicit `Confirm` parameter to this function it would be passed on to `Get-GitHubRepositoryBranch` which would then reprompt.
Once #254 is merged, both of these (`WhatIf` and `Confirm`) can be removed from this call.